src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py

```diff
@@ -0,0 +1,172 @@
+import sys
+import os
+import pytest
+from unittest import TestCase
+import yang as ly
+import sonic_yang as sy
+import json
+import getopt
+import subprocess
+
+test_path = os.path.dirname(os.path.abspath(__file__))
+modules_path = os.path.dirname(test_path)
+sys.path.insert(0, modules_path)
+
+class Test_SonicYang(object):
+    @pytest.fixture(autouse=True, scope='class')
+    def data(self):
+        test_file = "/sonic/src/sonic-yang-mgmt/tests/libyang-python-tests/test_SonicYang.json"
+        data = self.jsonTestParser(test_file)
+        return data
+
+    @pytest.fixture(autouse=True, scope='class')
+    def yang_s(self, data):
+        yang_dir = str(data['yang_dir'])
+        data_file = str(data['data_file'])
+        yang_s = sy.sonic_yang(yang_dir)
+        yang_s.load_schema_modules(yang_dir)
+
+        root = yang_s.load_data_file(data_file)
+        return yang_s
+
+    def jsonTestParser(self, file):
+        """
+        Open the json test file
+        """
+        with open(file) as data_file:
+            data = json.load(data_file)
+        return data
+
+    def setup_class(cls):
+        pass
+
+    def load_yang_model_file(self, yang_s, yang_dir, yang_file, module_name):
```

Comment:
I feel, we should write each function in try block and have a meaningful msg in except block as below:

try:
......
.......
except Exception as e:
        print("Yang Models Load failed")
        raise e

Reply:
Error messages added